### PR TITLE
Write function that loops through following week's recipes and compares to current pantry items. Adds non-existent pantry items to a list to remind the user of what's necessary to cook the upcoming recipes :: IN PROCESS

### PR DESCRIPTION
**Pull request by JayRodrig**
# Description
It's necessary to have a function that will look ahead of time and see what the necessary ingredients will be, compare these to the existing ingredients on the user's virtual pantry and tell the user what they actually need

# Problem
Until now we didn't have a function that solved that issue. User didn't have a to refer to when they needed to purchase their pantry items

# Solution
`getAllWeeklyMeals` _temporary name_ gets all the user's for the upcoming week, also gets all the items on their virtual pantry and it compares to see what they have and what they don't. Whatever they don't have is added to the shopping list

### CODE REVIEW

File: services/currentPantry.js [hunk 1]
```diff
@@ -44,10 +44,13 @@ const getPantryItemByName = name => getDbConn(dbAddr).any(
 const getPantryItemsOfUser = id => getDbConn(dbAddr).any(
```
Comment (as1729) [inline services/currentPantry.js]: rename this `id` to `userId` or `ownerId` so you know that its not the primary key of the `currentPantry` table.

File: services/ingredient.js [hunk 2]
```diff
@@ -47,10 +47,13 @@ const getIngredientByName = name => getDbConn(dbAddr).any(
 const getRecipeIngredients = id => getDbConn(dbAddr).any(
```
Comment (as1729) [inline services/ingredient.js]: another naming recommendation. This should ideally be the `recipeId` instead of `id` since you're querying by `recipe_id`

Comment (JayRodrig) [reply services/ingredient.js]: Perfect, will change it!

File: services/currentPantry.js (hunk 1 above)
Comment (JayRodrig) [reply services/currentPantry.js]: Perfect, will change it!

File: services/shopping_list.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    A DIFF ONE ON THE RECIPE'S ING
+
+    GIVEN THE OBS UP TOP, CONSIDER MAKING THE INGREDIENTS NAME UNIQUE
+    THAT WAY 2 'RICE' ROWS WON'T BE MADE
+    USERS WILL HAVE TO NAME THEM DIFF SINCE THEY'LL REPRESENT DIFF PRODUCTS
+*/
+
+
+const getAllWeeklyMeals = async _ => {
+    const weekMealsCall = await axios.get('http://localhost:11235/mealSchedule/user/1');
```
Comment (as1729) [inline services/shopping_list.js]: as we discussed try using `MealScheduleServices.getScheduledMeals(1)` directly instead of making another web request

Comment (as1729) [reply services/shopping_list.js]: You may need to figure out how to import those services into your file.

File: services/shopping_list.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    OR IF USING A CONSISTENT PROD FOR, LET'S SAY 'RICE', SPECIFY WHEN THEY ACTUALLY WANT 
+    A DIFF ONE ON THE RECIPE'S ING
+
+    GIVEN THE OBS UP TOP, CONSIDER MAKING THE INGREDIENTS NAME UNIQUE
+    THAT WAY 2 'RICE' ROWS WON'T BE MADE
+    USERS WILL HAVE TO NAME THEM DIFF SINCE THEY'LL REPRESENT DIFF PRODUCTS
+*/
+
+
+const getAllWeeklyMeals = async _ => {
```
Comment (as1729) [inline services/shopping_list.js]: I think this function is returning a list of ingredients the user needs to buy for the upcoming week. If that's the case a function name along the lines of - `ingredientsListForWeek` or `requiredIngredientsForWeeklyMeal` may be more appropriate.

File: services/shopping_list.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    };
+
+    let necessaryIng = [];
+    for (let recipe of weekRecipes) {
+        const ingCall = await axios.get(`http://localhost:11235/ingredient/recipe/${recipe.recipe_id}`);
+        const {data: recipeIng,} = ingCall.data;
+        necessaryIng = necessaryIng.concat(recipeIng);
+    };
+
+    let list = {};
```
Comment (as1729) [inline services/shopping_list.js]: would recommend being more explicit of what this list is. Like `ingredientsList`?

**Pull request merged**